src/lazy/expanded/macro_evaluator.rs

```diff
@@ -718,7 +712,7 @@ impl<'top, D: Decoder> MacroEvaluator<'top, D> {
 
     #[inline(never)]
     pub fn push_general_case(&mut self, new_expansion: MacroExpansion<'top, D>) {
-        match self.state {
+        match mem::take(&mut self.state) {
```

Comment:
🪧 An indirect use of `Copy`: because `self.state` isn't `Copy` any more, we need to set its value to the default (`Empty`) so we can update its state using the (owned) contents.